backend/app/services/user_info.rb

```diff
@@ -26,7 +43,74 @@ def self.for_uuids(uuids)
     user_uuids
   end
 
+  def self.for_uuid(uuid)
+    return dev_user_info(uuid) unless Rails.env.production?
+
+    for_uuids([uuid]).values[0]&.table || {}
+  end
+
   def self.query_accounts(query)
     OpenStax::Accounts::Api.search_accounts(query).response.body
   end
+
+  def self.dev_researchers
+    Researcher.all.map do |researcher|
+      {
+        user_id: researcher.user_id,
+        first_name: researcher.first_name,
+        last_name: researcher.last_name,
+        name: "#{researcher.first_name} #{researcher.last_name}",
+        isResearcher: true,
+        isAdmin: false
+      }
+    end
+  end
+
+  def self.dev_admins
+    Admin.all.map do |admin|
+      {
+        user_id: admin.user_id,
+        first_name: 'Admin',
+        last_name: 'McAdminFace',
+        name: 'Admin McAdminFace',
+        isAdmin: true,
+        isResearcher: false
+      }
+    end
+  end
```

Comment:
a few things:
* ruby uses underscore, not camel case so it should be `is_admin` etc.
* you shouldn't load all records into memory (what if there was 1000 researchers?)  just to find a single one.  Instead make methods for the properties that are not db fields, like:



```ruby
class Researcher < ApplicationRecord
  def is_researcher
    true 
   end
```

Then later to find a given researcher, call:

`Researcher.find_by(uuid: uuid)&.to_json(methods: [:is_admin, :is_researcher])`

Reply:
RE: underscores, i didn't even notice this, these were all copy pasted from the dev/users_controller so those must have been in here for a while 😨 my JS brain just glanced over it, good catch!

These two methods are for the dev users list, so we aren't actually loading all the researchers anywhere in reality, plus there will be a limited list locally.